### ISSUE TITLE
Add the Loss class

### ISSUE THREAD

**Issue by masa-su** [masa-su/pixyz#29]
VAEとか，任意のlossを実装できるようにしたいので，lossクラスを作って，それらで四則演算できるようにするといいかもしれない．
そうすると
1. 確率モデルの設計
2. 誤差関数の設計
を独立にできる．

通常の確率モデルは，1を設計した時点で2も決定するんだけど（尤度最大化など），深層生成モデルではlossをad-hocに設計することが多いので，このほうが嬉しい．

今はKullbackLeiblerなどがdivergenceの一つとして実装されている．これを全てLossクラスを継承する形に修正する．

例えば，こんな感じ
```
rec = Reconstruction(q, p)
div = KullbackLeibler(prior, q)
loss = rec - div
loss.estimate({"x": x})
```
